src/fs/constants.rs

```diff
@@ -5,6 +5,50 @@ use crate::backend;
 pub use crate::timespec::{Nsecs, Secs, Timespec};
 pub use backend::fs::types::*;
 
+impl FileType {
+    /// Returns `true` if this `FileType` is a regular file.
+    pub fn is_file(self) -> bool {
+        self == FileType::RegularFile
+    }
+
+    /// Returns `true` if this `FileType` is a directory.
+    pub fn is_dir(self) -> bool {
+        self == FileType::Directory
+    }
+
+    /// Returns `true` if this `FileType` is a symlink.
+    pub fn is_symlink(self) -> bool {
+        self == FileType::Symlink
+    }
+
+    /// Returns `true` if this `FileType` is a fifo.
+    #[cfg(not(target_os = "wasi"))]
+    pub fn is_fifo(self) -> bool {
+        self == FileType::Fifo
+    }
+
+    /// Returns `true` if this `FileType` is a socket.
+    #[cfg(not(target_os = "wasi"))]
+    pub fn is_socket(self) -> bool {
+        self == FileType::Socket
+    }
+
+    /// Returns `true` if this `FileType` is a character device.
+    pub fn is_char_device(self) -> bool {
+        self == FileType::CharacterDevice
+    }
+
+    /// Returns `true` if this `FileType` is a block device.
+    pub fn is_block_device(self) -> bool {
+        self == FileType::BlockDevice
+    }
+
+    /// Returns `true` if this `FileType` is unknown.
+    pub fn is_unknown(self) -> bool {
+        self == FileType::Unknown
+    }
```

Comment:
Is `is_unknown` useful? Rust's `FileTypeExt` [lacks it](https://doc.rust-lang.org/stable/std/os/unix/fs/trait.FileTypeExt.html#tymethod.is_char_device).

Reply:
Good point. I carry these functions as an extension trait for one or two years now and added the `is_unkown` only for "completeness" in this PR.

I've no problem without it.

Reply:
Makes sense; unless there's a specific need for it, my sense here is it's best to follow std.